controller/controller.go

```diff
@@ -275,6 +279,7 @@ func (c *Impl) processNextWorkItem() bool {
 		// Run Reconcile, passing it the namespace/name string of the
 		// resource to be synced.
 		if err = c.Reconciler.Reconcile(ctx, key); err != nil {
+			c.handleErr(err, key)
```

Comment:
Based on the comment below, I didn't think requeues were necessary (until `Forget` is explicitly called)

Reply:
The key point is that it does not behavior as it commented, as I described in knative/serving#2350, the key will not get requeued  automatically, unless we call `AddRateLimited` explicitly.
as a validation, you can scale up the sleep period [here in the test](https://github.com/knative/pkg/blob/master/controller/controller_test.go#L449), and you can see `NumRequeues` did not grow accordingly, which means the key is not indeed requeued and step into the reconcile look again.

Reply:
I guess the code here might be originally borrowed from sample-controller in k8s.
The behavior here has been fixed in the functional controllers. but no one noticed this in the sample one I think

Reply:
Cool, thanks for explaining